modin/pandas/io.py

```diff
@@ -799,6 +1168,15 @@ def return_handler(*args, **kwargs):
 class ExcelFile(pandas.ExcelFile):  # noqa: PR01, D200
     """
     Class for parsing tabular excel sheets into DataFrame objects.
+
+    Parameters Support Status:
+
+    +-----------------+-----------------+----------------+----------------+----------------------------------+
+    | Parameters      | PandasOnRay     | PandasOnDask   | OmniSci        | Notes                            |
+    +=================+=================+================+================+==================================+
+    | All parameters  | Harmful         | Harmful        | Harmful        | Please use `pd.read_excel`       |
```

Comment:
I actually didn't check content of all tables but this one glares. Why should `pd.read_excel` be used?

Reply:
`pd.read_excel` should be used instead because `ExcelFile` usage doesn't involve any computations parallization and it's usage can be harmful for user's application performance.

Reply:
Should we create an issue to make `ExcelFile` perform parallelization?

Reply:
`ExcelFile` seems to me slightly outdated tool and in my opinion `pd.read_excel` is more obvious option to use for excel files import, so it's fine to leave just compatible with pandas `ExcelFile` class. Probably we can add warning with note that `pd.read_excel` usage is preffered. What do you think?

Reply:
Yes, a warning would be enough for now.

Reply:
Ok, issue was added https://github.com/modin-project/modin/issues/4465.

Reply:
Thanks!